### ISSUE TITLE
CA1822 and IDE0052 is reported for private readonly ILogger<Foo> _Logger;

### ISSUE THREAD

**Issue by FlorianGrimm** [dotnet/roslyn-analyzers#6061]
### Analyzer

**Diagnostic ID**: [CA1822](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822)

### Describe the improvement

A method with LoggerMessageAttribute is reported for CA1822, which genererated version uses the this._Logger field.

```
IDE0052 Remove unread private member
private readonly ILogger<Foo> _Logger;

    [LoggerMessage(
        EventId = 123,
        Level = LogLevel.Trace,
        Message = "foo {a} {b};")]
    partial void LogGetDivision(int a, int b);
CA1822 "Make static"
IDE0060 Remove unused parameter
```

### Describe suggestions on how to achieve the rule

Handle non static version for LoggerMessage.

P.S.
One rules for CA and one two IDE.. I don't where to put it...

**Comment by mavasani** [comment]
I think this was recently fixed by @Youssef1313 with https://github.com/dotnet/roslyn-analyzers/pull/5172. This should be part of the .NET7 SDK (not sure if this made into .NET6 SDK as well). @Youssef1313 do you know?

**Comment by mavasani** [comment]
@Youssef1313 The issue description also mentions IDE0060 false positive. Do you recall fixing the false positive for that analyzer as well?

**Comment by Youssef1313** [comment]
@Youssef1313 I see the fix in the release/6.0.1xx branch

https://github.com/dotnet/roslyn-analyzers/blob/release/6.0.1xx/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs#L198-L200

So I think it should be included in .NET 6 SDK?

For IDE0052 and IDE0060, I have no idea.

**Comment by mavasani** [comment]
Thanks @Youssef1313 - closing this issue.
